cloud/amazon/cloudfront.py

```diff
@@ -0,0 +1,557 @@
+            },
+    }
+
+    if module.params.get('type') == "invalidation":
+        module.fail_json(msg="Invalidations cannot be updated or removed")
+    elif not module.params.get('resource_id'):
+        module.fail_json(msg="resource_id is requried for removing a resource")
+    elif module.params.get('wait_for_deployed'):
+        status_achieved, config = disable_distribution(client, module)
+    else:
+        invocation = invocations[module.params.get('type')]['method']
+        config = invocation(Id=module.params.get('resource_id'))
+
+    if not config[invocations[module.params.get('type')]['config_param']]['Enabled']:
+        resource_id = module.params.get('resource_id')
+        etag = config['ETag']
+        changed, result = removal(client, module, resource_id, etag)
+    else:
+        module.fail_json(msg="Resource must be disabled before you can remove it from AWS")
+
+    return changed, result
+
+
+def removal(client, module, resource_id, etag):
+    changed = False
+    params = dict()
+
+    invocations = {
+        'distribution': client.delete_distribution,
+        'origin_access_id': client.delete_cloud_front_origin_access_identity,
+        'streaming': client.delete_streaming_distribution
+    }
+
+    params['Id'] = resource_id
+    params['IfMatch'] = etag
+
+    invocation = invocations[module.params.get('type')]
+    result = json.loads(json.dumps(invocation(**params), default=date_handler))
+    changed = True
+    return (changed, result)
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        type=dict(choices=[
+            'distribution',
+            'origin_access_id',
+            'invalidation',
+            'streaming',
+        ], required=True),
+        policy=dict(),
+        resource_id=dict(),
+        state=dict(default='present', choices=['present', 'absent']),
+        wait_for_deployed=dict(type='bool', default=False),
+        )
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+    )
+
+    # Validate Requirements
+    if not HAS_BOTO3:
+        module.fail_json(msg='json and botocore/boto3 is required.')
+
+    state = module.params.get('state').lower()
+    #Cloudfront is non-region specific - default global region to us-east-1/US Standard
+    module.params['region'] = 'us-east-1'
+
+    try:
+        region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+        cloudfront = boto3_conn(module, conn_type='client', resource='cloudfront', region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except Exception as e:
+        module.fail_json(msg="Can't authorize connection - "+str(e))
+
+    #Ensure resource is present
+    if state == 'present':
+        (changed, results) = creation_setup(cloudfront, module)
+    else:
+        (changed, results) = removal_setup(cloudfront, module)
+
+    module.exit_json(changed=changed, result=results)
```

Comment:
Make sure any results are camel_cased.  Use the module libs ec2 helper function camel_dict_to_snake_dict (see GUIDELINES)